prow/kube/config_test.go

```diff
@@ -109,3 +115,241 @@ func TestMergeConfigs(t *testing.T) {
+  name: build01
+- context:
+    cluster: api-build02-gcp-ci-openshift-org:6443
+    namespace: hongkliu-test
+    user: hongkailiu/api-build02-gcp-ci-openshift-org:6443
+  name: build02
+- context:
+    cluster: api-hive-9xw5-p1-openshiftapps-com:6443
+    namespace: hongkliu-test
+    user: hongkliu/api-hive-9xw5-p1-openshiftapps-com:6443
+  name: hive
+current-context: app.ci
+kind: Config
+preferences: {}
+users:
+- name: hongkailiu/api-build01-ci-devcluster-openshift-com:6443
+  user:
+    token: REDACTED
+- name: hongkailiu/api-build02-gcp-ci-openshift-org:6443
+  user:
+    token: REDACTED
+- name: hongkailiu/api-ci-l2s4-p1-openshiftapps-com:6443
+  user:
+    token: REDACTED
+- name: hongkliu/api-hive-9xw5-p1-openshiftapps-com:6443
+  user:
+    token: REDACTED`
+
+	b01AndHive = `apiVersion: v1
+clusters:
+- cluster:
+    server: https://api.build01.ci.devcluster.openshift.com:6443
+  name: api-build01-ci-devcluster-openshift-com:6443
+- cluster:
+    server: https://api.hive.9xw5.p1.openshiftapps.com:6443
+  name: api-hive-9xw5-p1-openshiftapps-com:6443
+contexts:
+- context:
+    cluster: api-build01-ci-devcluster-openshift-com:6443
+    namespace: hongkliu-test
+    user: hongkailiu/api-build01-ci-devcluster-openshift-com:6443
+  name: build01
+- context:
+    cluster: api-hive-9xw5-p1-openshiftapps-com:6443
+    namespace: hongkliu-test
+    user: hongkliu/api-hive-9xw5-p1-openshiftapps-com:6443
+  name: hive
+current-context: build01
+kind: Config
+preferences: {}
+users:
+- name: hongkailiu/api-build01-ci-devcluster-openshift-com:6443
+  user:
+    token: REDACTED
+- name: hongkliu/api-hive-9xw5-p1-openshiftapps-com:6443
+  user:
+    token: REDACTED`
+
+	b02AndAPPCI = `apiVersion: v1
+clusters:
+- cluster:
+    server: https://api.build02.gcp.ci.openshift.org:6443
+  name: api-build02-gcp-ci-openshift-org:6443
+- cluster:
+    server: https://api.ci.l2s4.p1.openshiftapps.com:6443
+  name: api-ci-l2s4-p1-openshiftapps-com:6443
+contexts:
+- context:
+    cluster: api-ci-l2s4-p1-openshiftapps-com:6443
+    namespace: hongkliu-test
+    user: hongkailiu/api-ci-l2s4-p1-openshiftapps-com:6443
+  name: app.ci
+- context:
+    cluster: api-build02-gcp-ci-openshift-org:6443
+    namespace: hongkliu-test
+    user: hongkailiu/api-build02-gcp-ci-openshift-org:6443
+  name: build02
+current-context: build02
+kind: Config
+preferences: {}
+users:
+- name: hongkailiu/api-build02-gcp-ci-openshift-org:6443
+  user:
+    token: REDACTED
+- name: hongkailiu/api-ci-l2s4-p1-openshiftapps-com:6443
+  user:
+    token: REDACTED`
+)
+
+func TestLoadClusterConfigs(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatalf("failed to craete a temp directory: %v", err)
+	}
+	configFile := filepath.Join(dir, "all-config")
+	if err := ioutil.WriteFile(configFile, []byte(all), 0644); err != nil {
+		t.Fatalf("failed to write kubeconfig file %q: %v", configFile, err)
+	}
+	configDir := filepath.Join(dir, "some-dir")
```

Comment:
This is static test data. Either inline it into the test case: add a field to the test case struct `func() error` or whatever that generates the files, or add the files to test data directories.

Reply:
Updated.